components/sup/src/manager/service/hook_runner.rs

```diff
@@ -40,7 +57,28 @@ impl<H> HookRunner<H> where H: Hook + Sync
                      pkg,
                      passwd }
     }
+
+    // TODO (DM): Make `HookRunner` and this method allow arbirary configuration of how to run a
+    // hook. For example, a configurable backoff, timeout, or failure tracking. It would also be
+    // nice to merge this with the healthcheck hook retry logic.
```

Comment:
Just a note to make sure this TODO is resolved before merge

Reply:
Personally, this sounds a bit like YAGNI. Do we have a use case that demands this today? If not, I'd leave it out.

Reply:
I agree. I will remove this comment. I wanted a way to signify to reviewers that this is where we could add this functionality and get their feedback. In that way the comment has served its purpose :smile: . Maybe a better method would have been to just leave a comment on the PR?

I do think it would be useful to merge this logic with the health check retry logic, and there is an [issue](https://github.com/habitat-sh/habitat/issues/6701) about adding an initial delay before starting a service and many of those other features are currently TODOs in other parts of the code. Ultimatly, I think we will end up implementing some of this logic, but it is outside the scope of this PR.